subtests/docker_cli/login/login.py

```diff
@@ -161,6 +161,9 @@ def docker_login(self, password=None):
         """
         if password is None:
             password = self.sub_stuff['passwd']
+        # FIXME: --email option is deprecated in 1.12, and will be
+        # removed in 1.13. We still need it for <= 1.10; otherwise
+        # docker prompts from stdin, and we hang.
```

Comment:
What I meant was more along the lines of: less sloppy checking. Instead of "does output contain the substring X", something like "is output a single line of the form <anchored-regex>". That way new warnings will not be ignored.